libraries/freertos_plus/aws/ota/src/http/aws_iot_ota_http.c

```diff
@@ -435,6 +436,12 @@ static void _httpReadReadyCallback( void * pPrivateData,
     /* Buffer to read the "Connection" field in HTTP header. */
     char connectionValueStr[ HTTP_HEADER_CONNECTION_VALUE_MAX_LEN ] = { 0 };
 
+    /* Bail out if this callback is invoked after http downloader stopped. */
+    if( _httpDownloader.state == OTA_HTTP_STOPPED )
+    {
+        return;
```

Comment:
```suggestion
        IotHttpsClient_CancelResponseAsync( responseHandle );
        return;
```

Reply:
But this will invoke the errorcallback saying saying the user cancelled.

Reply:
Per offline conversation `IotHttpsClient_CancelRequestAsync( requestHandle );` will also be added to _httpAppendHeaderCallback().